core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -92,6 +94,14 @@ private static void toJson(Types.StructType struct, Integer schemaId, Set<Intege
       if (field.doc() != null) {
         generator.writeStringField(DOC, field.doc());
       }
+      if (field.initialDefaultValue() != null) {
+        generator.writeFieldName(INITIAL_DEFAULT);
+        generator.writeRawValue(field.initialDefaultValue().toString());
+      }
+      if (field.writeDefaultValue() != null) {
+        generator.writeFieldName(WRITE_DEFAULT);
+        generator.writeRawValue(field.writeDefaultValue().toString());
```

Comment:
Instead of using `toString` that produces JSON, I think that this should expose a `ValueParser` that converts values from the internal Iceberg representation into JSON values.